### PR TITLE
OCMUI-3288: Enhance HTPasswd username validation logic

### PR DESCRIPTION
**Pull request by zherman0**
# Summary
The UI and CLI did not match for htpasswd usernames.  Based on the htpasswd standards, spaces should be allowed so I have updated the code to allow spaces in the username or as a username - i.e. "     " (valid by standards)

- Added a maximum length check for HTPasswd usernames (255 characters).
- Updated validation messages for empty usernames and invalid characters.
- Adjusted unit tests to reflect new validation rules and edge cases.

# Jira

Fixes [OCMUI-3288](https://issues.redhat.com/browse/OCMUI-3288)

# Additional information

Refer to the jira ticket to see the CLI commands if you want to test the CLI

# How to Test

1.  Build an HCP cluster
2. When ready, go to cluster details-> "Access control" tab
3. On the sub menu "Identity providers", Add a htpasswd IDP
4. Validate that the username can be spaces or have some like "test user 1".  Please realize that in an ASCII file " " (1 space_ is different thn "  " (2 spaces).  So you could have a real fun list of usernames.
5. Validate the username cannot be left empty
6. If you are bored, validate that you cannot have more than 255 characters.

# Screen shots
Valid usernames (2 spaces, 3 spaces, 4 spaces):
Screenshot From 2026-06-30 15-51-22

Username with spaces:
Screenshot From 2026-06-30 15-53-30

Username
Screenshot From 2026-06-30 15-53-50
 with more than 255 chars:

# Review process

Please review and follow the [PR process](https://github.com/RedHatInsights/uhc-portal/blob/main/docs/pull-request-process.md).

## QE Reviewer

- [ ] _Pre-merge testing : Verified change locally in a browser (downloaded and ran code using reviewx tool)_
- [ ] Updated/created Polarion test cases which were peer QE reviewed
- [ ] Confirmed 'tc-approved' label was added by dev to the linked JIRA ticket
- [ ] (optional) Updated/created Cypress e2e tests
- [ ] Closed threads I started after the author made changes or added an explanation

[OCMUI-3288]: https://redhat.atlassian.net/browse/OCMUI-3288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jmekkatt — CHANGES_REQUESTED** [review]
Hi @zherman0 , An e2e test spec [identity-provider-htpasswd-manual.spec.ts](https://github.com/RedHatInsights/uhc-portal/blob/main/playwright/e2e/access-control/identity-provider-htpasswd-manual.spec.ts)  influence with this change. We would need to add and update the test coverage based on this new change. Could you please help to cover the same? Thanks

**Comment by jmekkatt** [comment]
Hi @zherman0 , Could you please review this [comment](https://redhat.atlassian.net/browse/OCMUI-3288?focusedCommentId=17454098) on multiple whitespace support for username field?
If we decide to support usernames consisting only of whitespace, we should also support the same behavior when users upload credentials via "Upload an htpasswd file". Otherwise, the behavior would be inconsistent between the two methods.

**Comment by jmekkatt** [comment]
Hi @zherman0 ,
The new change i.e. you cannot have more than 255 characters in username not respected when users upload credentials via "Upload an htpasswd file". Could you please fix the same? Thank you.

**Comment by zherman0** [comment]
@jmekkatt - I know there are still general concerns for this change, but I did address and fix all of your mentioned items so e2e pass and the file handles spaces/255 max.

**Review by DcooperRH — APPROVED** [review]
Tested various empty space scenarios for the usernames. Gotta say, I don't love the ux of allowing these usernames, but I understand why we are. 
Good unit tests, LGTM

**Comment by zherman0** [comment]
Waiting on feedback from the PM to see how to proceed on allowing blank user names

File: src/components/clusters/ClusterDetailsMultiRegion/components/IdentityProvidersPage/components/ProvidersForms/htpasswdFileParser.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { validateHTPasswdUsername } from '~/common/validators';
```
Comment (lizagilman) [inline src/components/clusters/ClusterDetailsMultiRegion/components/IdentityProvidersPage/components/ProvidersForms/htpasswdFileParser.ts]: Nit: Mind moving `validateHTPasswdUsername` closer to the consumers? It's not really "common" (/ shared) with other components

Comment (zherman0) [reply src/components/clusters/ClusterDetailsMultiRegion/components/IdentityProvidersPage/components/ProvidersForms/htpasswdFileParser.ts]: Good idea, the validators file is far too big as is.  Created a new Providers helper file in the ProvidersForms directory.

**Comment by zherman0** [comment]
@jmekkatt - I have updated this PR to basically only allow spaces between characters, but no spaces leading or trailing which means all spaces in no longer allowed.  Please see the notes in the Jira ticket.

File: playwright/fixtures/access-control/identity-provider-htpasswd.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
       "Usernames": {
         "EmptyUserNameError": "Username is required",
         "InvalidUserNameInput": ["Osd%Rosa:asd", "1234567890"],
-        "InValidUserNameError": "Username must not contain /, :, %, or empty spaces.",
+        "InValidUserNameError": "Username must not contain /, :, or %.",
```
Comment (jmekkatt) [inline playwright/fixtures/access-control/identity-provider-htpasswd.json:15]: Hi @zherman0 , 
E2E tests do not cover the two new validation rules (leading/trailing spaces, 255-char limit)
- Entering " username" or "username " → expected error: "Username must not contain leading or trailing spaces."
- Entering a 256+ character username → expected error: "Username may not exceed 255 characters."
Could you please  update the same?

**Comment by zherman0** [comment]
@jmekkatt -
1) Updated the E2E to have additional checks for leading/trailing spaces and 255+ characters.
2) Improved consistency - we now throw an error for usernames with leading/trailing spaces in the file upload as well (which is how it works currently in prod) but also matches the behavior in the single input UI
